src/DataService/Batch.php

```diff
@@ -544,6 +564,16 @@ private function ProcessBatchItemResponse($oneXmlObj)
               case "Fault":
                   $result->responseType = UtilityConstants::Exception;
                   $idsException = $this->IterateFaultAndPrepareException($firstChild);
+                  if($this->debugMode) {
```

Comment:
What is the rationale for the debug mode? We do capture the logs in L371 in case of an exception right? That has the body and header information?

Reply:
The debug mode is needed because you need to extract the debug information from the response itself. And then inject this into the exception which you can extract. Code example below.